daphne_worker/src/dap.rs

```diff
@@ -595,6 +595,19 @@ where
     }
 }
 
+fn task_id_from_report(report: &[u8]) -> std::result::Result<Id, DapError> {
+    // The task id is the first 32 bytes of the serialized report.
```

Comment:
This is true for drafts 02 and 03, but it may change in the future. To prevent ourselves from shooting ourselves in the foot later, it might be useful to fence this method to those versions somehow. Maybe pass the version here and err unless we call on a compatible version?

Reply:
We can't do that, as we're calling this function to get the task id, so we can find the task, and thus learn the version.  I strengthened the comment to note that if we want to change the layout in the future, we have to change the serialization form of the DO so that it has an explicit version.